diff --git a/common/cmd_aml.c b/common/cmd_aml.c
index 58084e7..11d2f81 100755
--- a/common/cmd_aml.c
+++ b/common/cmd_aml.c
@@ -18,7 +18,7 @@ extern inline int is_ac_online(void);
 extern void power_off(void);
 extern inline int get_charging_percent(void);
 extern inline int set_charging_current(int current);
-
+extern void rn5t618_chg_led_control(int battery_percent);
 
 static int do_getkey (cmd_tbl_t *cmdtp, int flag, int argc, char * const argv[])
 {
@@ -109,6 +109,7 @@ static int do_get_batcap (cmd_tbl_t *cmdtp, int flag, int argc, char * const arg
 	printf("Battery CAP: %d%%\n", percent);
 	sprintf(percent_str, "%d", percent);
 	setenv("battery_cap", percent_str);
+    rn5t618_chg_led_control(percent);
 	return 0;
 }
 
