From 136b70e1b974cc25e8951e8f5299f981ec7e5c2b Mon Sep 17 00:00:00 2001
From: "joy.wang" <joy.wang@amlogic.com>
Date: Tue, 3 Jun 2014 16:54:36 +0800
Subject: [PATCH] DCDC2 frequent set to 2.2MHz for seven

---
 drivers/power/rn5t618.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/power/rn5t618.c b/drivers/power/rn5t618.c
index 36ac8bc..b336de5 100755
--- a/drivers/power/rn5t618.c
+++ b/drivers/power/rn5t618.c
@@ -666,6 +666,7 @@ int rn5t618_init(void)
     rn5t618_set_bits(0x002C, 0x10, 0x30);                       // DCDC1 set to PWM mode
     rn5t618_set_bits(0x002E, 0x10, 0x30);                       // DCDC2 set to PWM mode
     rn5t618_set_bits(0x0030, 0x10, 0x30);                       // DCDC3 set to PWM mode
+	rn5t618_set_bits(0x002F, 0x40, 0xc0);                       // DCDC2 frequent set to 2.2MHz
     rn5t618_set_gpio(3, 0);                                     // open GPIO 2, DCDC 3.3
     rn5t618_set_gpio(1, 0);                                     // close GPIO 1, boost 5V 
     rn5t618_set_bits(0x00EF, 0x10, 0x10);                       // enable coulomb counter
-- 
1.7.9.5

